samples/extensions/shader_debugprintf/shader_debugprintf.cpp

```diff
@@ -393,9 +416,37 @@ bool ShaderDebugPrintf::prepare(const vkb::ApplicationOptions &options)
 	return true;
 }
 
+// This sample overrides the per-sample layer framework to force activation of the validation layer
+const std::vector<const char *> ShaderDebugPrintf::get_validation_layers()
+{
+	// Validation layer is already enabled for debug builds, so initialize override list to default (empty)
+	std::vector<const char *> validation_layers = ApiVulkanSample::get_validation_layers();
+
+#if !defined(VKB_DEBUG) && !defined(VKB_VALIDATION_LAYERS)
+	// Force activation of validation layer on release builds for access to debugPrintfEXT feature
+	validation_layers.push_back("VK_LAYER_KHRONOS_validation");
+#endif
+
+	return validation_layers;
+}
+
 // This sample overrides the instance creation part of the framework to chain in additional structures
 std::unique_ptr<vkb::Instance> ShaderDebugPrintf::create_instance(bool headless)
 {
+	uint32_t instance_extension_count;
+	VK_CHECK(vkEnumerateInstanceExtensionProperties(nullptr, &instance_extension_count, nullptr));
+	std::vector<VkExtensionProperties> available_instance_extensions(instance_extension_count);
+	VK_CHECK(vkEnumerateInstanceExtensionProperties(nullptr, &instance_extension_count, available_instance_extensions.data()));
+
+	// When VK_EXT_layer_settings is available at runtime, the debugPrintfEXT layer feature is enabled using the standard framework
```

Comment:
Can you elaborate on this? I read through that comment multiple times but I still don't get why it's needed. If it's really required please also add a note to the sample's readme.

Reply:
This runtime check for `VK_EXT_layer_settings` allows both `create_instance()` approaches to coexist and not break backwards compatibility for older SDKs or on platforms that don't yet support `VK_EXT_layer_settings`.  If backwards compatibility is not a concern, then the `create_instance()` override method would not be needed at all, and it could be removed from the code for 100% reliance on `VK_EXT_layer_settings` to enable debug printf.

Given that layer settings is relatively new, I felt that backwards compatibilty in this case was necessary.

Reply:
If you are okay with this dual track approach for backwards compatability in this sample, I will come back and add info to the README explaining it.  Please let me know.

Reply:
Totally fine with it 👍🏻

Reply:
Thanks - I will update the README tonight and close this final issue before requesting a final re-review.

Reply:
README updated.  Closing this specific comment which resolves the last outstanding issue.  I will now request re-reviews of the PR, hopefully we are getting closer!